src/libostree/ostree-repo-pull-private.h

```diff
@@ -72,7 +72,11 @@ typedef struct {
   gboolean          has_tombstone_commits;
 
   GBytes           *summary_data;
+  char             *summary_etag;
+  guint64           summary_last_modified;  /* seconds since the epoch */
```

Comment:
I think `ETag` support is actually more important than `Last-Modified` support because it doesn’t have race conditions around updates/checks in the same second, and doesn’t have issues with clock synchronisation. Both require some level of support on the server (typically, aiui, servers are configured to serve everything as if it’s not cacheable, as that’s the safest option). I would expect that all web servers would support both headers, though. They’ve been around a long time.

If xattr support isn’t available then ETags won’t work, but should degrade nicely, so that shouldn’t cause a problem.